src/main/java/com/thealgorithms/scheduling/HighestResponseRatioNextScheduling.java

```diff
@@ -0,0 +1,140 @@
+package com.thealgorithms.scheduling;
+
+import java.util.Arrays;
+import java.util.Comparator;
+
+/**
+ * The HighestResponseRatioNextScheduling class implements
+ * the Highest Response Ratio Next (HRRN) scheduling algorithm.
+ * HRRN is a non-preemptive scheduling algorithm that
+ * selects the process with the highest response ratio for execution.
+ * The response ratio is calculated as (waiting time + burst time) / burst time.
+ * This algorithm aims to reduce the average waiting time
+ * and improve overall system performance by balancing short and long processes.
+ */
+public final class HighestResponseRatioNextScheduling {
+    private HighestResponseRatioNextScheduling() {
+    }
+
+    /**
+     * Calculates the Turn Around Time (TAT) for each process.
+     *
+     * @param processNames Array of process names.
+     * @param arrivalTimes Array of arrival times corresponding to each process.
+     * @param burstTimes   Array of burst times for each process.
+     * @param noOfProcesses The number of processes.
+     * @return An array of Turn Around Times for each process.
+     */
+    public static int[] calculateTurnAroundTime(String[] processNames, int[] arrivalTimes, int[] burstTimes, int noOfProcesses) {
```

Comment:
Some of these parameters in another method may be final to emphasize that they are constant within the method.

Reply:
Noted, I have changed the function header to the following:

```java
public static int[] calculateTurnAroundTime(final String[] processNames, final int[] arrivalTimes, final int[] burstTimes, final int noOfProcesses)
```